sycl/source/detail/scheduler/scheduler.cpp

```diff
@@ -141,6 +141,19 @@ EventImplPtr Scheduler::addHostAccessor(Requirement *Req) {
 
 void Scheduler::releaseHostAccessor(Requirement *Req) {
   Req->MBlockedCmd->MCanEnqueue = true;
+  MemObjRecord* Record = Req->MSYCLMemObj->MRecord.get();
+  for (Command *Cmd : Record->MReadLeafs) {
+    EnqueueResultT Res;
+    bool Enqueued = GraphProcessor::enqueueCommand(Cmd, Res);
+    if (!Enqueued && EnqueueResultT::FAILED == Res.MResult)
+      throw runtime_error("Enqueue process failed.");
+  }
+  for (Command *Cmd : Record->MWriteLeafs) {
+    EnqueueResultT Res;
+    bool Enqueued = GraphProcessor::enqueueCommand(Cmd, Res);
+    if (!Enqueued && EnqueueResultT::FAILED == Res.MResult)
+      throw runtime_error("Enqueue process failed.");
+  }
```

Comment:
You could factorize the code in a lambda and call it on `Record->MReadLeafs` and `Record->MWriteLeafs`